### PR TITLE
[7.x] Add ArrayAccess support for Http client get requests

### LINKED ISSUE #32398: Http::assertSent on get request causes Undefined Index exception
- Laravel Version: 7.6.2
- PHP Version: 7.4.4

### Description:
When doing an `Http::assertSent` I am getting `Undefined index: Content-Type` when testing a `Http::get` request that wouldn't have a `Content-Type`.

```bash
ErrorException : Undefined index: Content-Type
   .../Illuminate/Http/Client/Request.php:72
   .../Illuminate/Http/Client/Request.php:182
   .../Illuminate/Http/Client/Request.php:136
   .../Illuminate/Http/Client/Request.php:237
  // ... line on my test
```

### Steps To Reproduce:

Here is an example that is similar to my actual code:

```php
public function get(): array
{
    return \Http::acceptJson()
        ->get('https://google.com', ['apikey' => 'aaa-bbb-ccc'])
        ->json();
}
```

Then in my test I have something like the below:

```php
public function testGet()
{
    \Http::fake(['*' => \Http::response(['response' =>'Test return data'], 200)]);

    // .. call get()

    \Http::assertSent(function (Request $request) {
        $this->assertStringStartsWith('https://google.com', $request->url());
        $this->assertSame('aaa-bbb-ccc', $request['apikey']); // Undefined index: Content-Type
        return true;
    });
}
```

### Cause
Looking at the line causing the error it seems it just needs to check if the key it is searching for exists in the headers array before trying to access it:

**Illuminate/Http/Client/Request.php:**
```php
public function isForm()
{
    return $this->hasHeader('Content-Type', 'application/x-www-form-urlencoded');
}

...

public function hasHeader($key, $value = null)
{
    ...
    // Needs an Arr::has or array_key_exists here? This ⬇️ is causing the error. 
    return empty(array_diff($value, $this->headers()[$key]));
}
```

### PR DESCRIPTION
**Pull request by dmason30**
Fixes https://github.com/laravel/framework/issues/32398.

**TL;DR**
Fixes an `Undefined index: Content-Type` exception and allows access to get request params via array access `$request['foo']` and in turn allows you to make assertions such as `$request['foo'] === 'bar'`.

### Description:
I was writing a test similar to the below and when running it was greeted with an `ErrorException` of `Undefined index: Content-Type`. 

```php
public function testGet()
{
    \Http::fake(['*' => \Http::response(['response' =>'Test return data'], 200)]);

    \Http::acceptJson()
        ->get('https://google.com', ['apikey' => 'aaa-bbb-ccc'])
        ->json()

    \Http::assertSent(function (Request $request) {
        $this->assertStringStartsWith('https://google.com', $request->url());
        $this->assertSame('aaa-bbb-ccc', $request['apikey']); // Undefined index: Content-Type
        return true;
    });
}
```

### Cause
Looking at the line causing the error it seems it just needs to check if the key it is searching for exists in the headers array before trying to access it:

**Illuminate/Http/Client/Request.php:**
```php
public function isForm()
{
    return $this->hasHeader('Content-Type', 'application/x-www-form-urlencoded');
}

public function hasHeader($key, $value = null)
{
    ...
    // Added an Arr::has($headers, $key) here. This ⬇️ was causing the error. 
    return empty(array_diff($value, $this->headers()[$key]));
}
```

### Initial fix & New error found
So I initially set out to fix `hasHeader` so that it checks the `$key` exists in the `array` returned by `$this->headers()` and I ran the tests and got the same error `Undefined index: Content-Type` but from a different location:

**Illuminate/Http/Client/Request.php:**
```php
public function isJson()
{
    return Str::contains($this->header('Content-Type')[0], 'json');
}

public function header($key)
{
    // Caused by this line ⬇️ 
    return $this->headers()[$key];
}
```

### Adding ArrayAccess Support
I then added `Arr::get` check in the `header($key)` method and then discovered that there is not any existing support for accessing get query parameters via `$request` and was met with another `Undefined index: foo` in the below test:

```php
public function testGetWithArrayQueryParam()
{
    $this->factory->fake();

    $this->factory->get('http://foo.com/get', ['foo' => 'bar']);

    $this->factory->assertSent(function (Request $request) {
        return $request->url() === 'http://foo.com/get?foo=bar'
            && $request['foo'] === 'bar'; // Undefined index: foo
    });
}
```

So this change gets the query parameters either from `$options['query']` or by extracting them from the `$url`. It handles the below with the tests adjusted to prove it. 

```php
Http::get('http://foo.com');
Http::get('http://foo.com?foo=bar');
Http::get('http://foo.com', ['foo' => 'bar']);
Http::get('http://foo.com', 'foo=bar');
Http::get('http://foo.com/get?foo=bar&page=1', ['hello' => 'world']); // Hello world overwrites params
Http::get('http://foo.com', ['foo;bar; space test' => 'laravel']);
```
Then you can do assertions on the `$request` data:

```php
Http::assertSent(function (Request $request) {
    return $request->url() === 'http://foo.com/get?foo=bar'
        && $request['foo'] === 'bar';
});
```

### CODE REVIEW

**Comment by driesvints** [comment]
@dmason30 probably best to explain why this is needed, not just link to an issue.

**Comment by dmason30** [comment]
@driesvints Thanks done as requested.
